### ISSUE TITLE
Feature: RoundedDrawable

### ISSUE THREAD

**Issue by AlexBlokh** [koush/ion#375]
It's way more efficient to have a rounded Drawable instead of making a transformation with creating a bitmap. So, I'm able to create an implementation of an IonDrawable and let the user to round corners or make it circle.. thanks

Just have to create 
``BitmapShader, RectF and LocalMatrix for shader``
and that's be very efficient

**Comment by schwiz** [comment]
Unfortunately manishkungwani  that solution can't be used because it isn't possible to use a custom drawable with ion it will replace your drawable with its own. Sadly the ion drawable is package private so you can't extend it to add this functionality either. This needs fixing bad.

**Comment by koush** [comment]
Thoughts on adding a clip path argument to image views? This would let you use a totally custom clip effect, including rounded rect.

http://stackoverflow.com/questions/1705239/how-should-i-give-images-rounded-corners-in-android

**Comment by schwiz** [comment]
Pretty good idea, however I still think it would be more preferable to be able to either extend IonDrawable or make it not required. Perhaps an api that lets us set a drawable to be wrapped inside of yours. That would make it easier to also add shadows to the rounded image pre-L.

**Comment by koush** [comment]
Well, the placeholders and errors need to be rounded too...

**Comment by koush** [comment]
Can't get rid of IonDrawable. It's necessary for a couple reasons:

fade in animation
but mainly, iit invokes the network call during onDraw, so only on screen elements are retrieved

**Comment by schwiz** [comment]
Presumably the user would pass their own rounded drawable into placeholder and error as well. The reasoning for needing IonDrawable makes sense.  

You could let IonDrawable optionally implement rounded drawables and shadows as well, or make it extendable with hooks in draw to allow people to call something other than canvas.drawBitmap

**Comment by koush** [comment]
Released

**Comment by koush** [comment]
https://github.com/koush/ion/blob/master/ion-sample/src/com/koushikdutta/ion/sample/RoundedImageViewSample.java

**Comment by schwiz** [comment]
Sadly, this update broke KenBurnsView https://github.com/flavioarfaria/KenBurnsView
Images are no longer the correct size using the builder.intoImageView with KenBurnsView as the target. Was working just fine in 2.0.7

**Comment by koush** [comment]
@schwiz Lemme check that out. I fixed some image size issues in the master branch.

**Comment by koush** [comment]
Seems fine here on 207 209 and master. Video?

**Comment by koush** [comment]
@schwiz Added a sample just to test.
7bb773c

Note that I'm using deepzoom for that kenburns, but it is not necessary. Worked for me without it.

**Comment by schwiz** [comment]
Thanks I'll check that sample out and see. Right now I can switch my issue on/off by switching for 207 to 209 but your sample may help. If not I can get you a video and try to get some reproducible code as well.

**Comment by koush** [comment]
You may also wanna try turning off .smartSize

**Comment by schwiz** [comment]
Well, the sample didn't really help. What I'm seeing is actually this problem which the author of KenBurnsView closed out because the person who reported it never got back to him.  https://github.com/flavioarfaria/KenBurnsView/issues/7

I'm not sure exactly what the problem, but what I do know is if I use deepzoom the problem also goes away. 

A few more details in case you are interested:
1.) The KenBurnsView is wrapped in a special FrameLayout which will always have the same aspect ratio.
2.)KenBurnsView acted funny inside of that frame layout untill I added a callback which calls restart and resume on the KenBurnsView after the image loads.

So in short, I think this KenBurnsView setup was fragile already and while the changes to IonDrawable brought out more issues it is probably not Ion's fault. Also, the good news is everything is working again now that I have enabled DeepZoom. So thanks for your help!

**Comment by koush** [comment]
Try disabling smart size, I bet that will fix it

**Comment by schwiz** [comment]
I wasnt using that unless its enabled by default. Wasnt aware of the
feature.

**Comment by schwiz** [comment]
Ok, something strange is definitely going on, not just with KenBurnsView. I though smartSize(false) into my builder just to be sure. 

I have ImageViews wrapped in a custom FixedRatioFrame that will insure the ImageView is the aspect ratio that I know my image will be. 

I have a web service that takes arguments for the size of the ImageView and returns a sized image the exact size of the ImageView.

Scaletype on ImageView is centerInside (for case when client doesn't use resize service if it had already downloaded an image that won't need to be scaled way up ie it already has a bigger copy of the image downloaded)

using builder.intoImageView API sometimes causes these images to be zoomed way in instead of fitting the imageview. Not everytime, and not ever if loading from memcache. This application has been in development for about 6 months and I have never seen the issue until updating to 209, now I see it frequently.
